library/core/src/ffi/c_str.rs

```diff
@@ -126,35 +126,20 @@ pub struct CStr {
 /// ```
 #[derive(Clone, PartialEq, Eq, Debug)]
 #[stable(feature = "core_c_str", since = "1.64.0")]
-pub struct FromBytesWithNulError {
-    kind: FromBytesWithNulErrorKind,
-}
-
-#[derive(Clone, PartialEq, Eq, Debug)]
-enum FromBytesWithNulErrorKind {
+pub enum FromBytesWithNulError {
+    /// Data provided contains an interior nul byte at byte pos `usize`.
     InteriorNul(usize),
```

Comment:
This should be changed to a name field with its own doc comment.

Reply:
@Amanieu could you give an example of some other enum that already does this?  I looked at it seems most enums use tuple values.  Also, what do you mean by its own doc comment?  Would this be sufficient?

```rust
/// Data provided contains an interior nul byte at byte `pos`.
InteriorNul { pos: usize },
```

Reply:
ah, nvm about the docs.  I amended my PR - see if that's what you meant